tests/integration/components/model-picker-test.js

```diff
@@ -52,330 +54,318 @@ moduleForComponent('model-picker', 'Integration | Component | {{model-picker}}',
+                                   searchProperty="name"
+                                   prompt="Prompt text"}}`);
 
-  this.render(hbs`{{model-picker updated=(action 'update')
-                                 model="foo"
-                                 optionLabelPath="name"
-                                 searchProperty="name"}}`);
+    assert.equal(findAll('.model-picker').length, 1,
+                 'renders with model-picker class');
+    assert.equal(find('.ember-power-select-placeholder').textContent.trim(),
+                 'Prompt text', 'renders with given prompt as placeholder');
+  });
 
-  clickTrigger();
+  test('Searches for given model by attribute', async function(assert) {
+    await render(hbs`{{model-picker updated=(action 'update')
+                                    model="foo"
+                                    optionLabelPath="name"
+                                    searchProperty="name"}}`);
 
-  let $input = $('.ember-power-select-dropdown input');
-  let $items = $('.ember-power-select-dropdown li');
+    await clickTrigger();
 
-  assert.equal($items.length, 1, 'by default no content');
-  assert.equal($items[0].innerText, 'Type to search');
+    let $items = $('.ember-power-select-options li');
 
-  run(() => {
-    $input.val('bar');
-    $input.trigger('input');
-  });
+    assert.equal($items.length, 1, 'by default no content');
+    assert.equal($items[0].innerText, 'Type to search');
 
-  wait().then(() => {
-    $items = $('.ember-power-select-dropdown li');
+    await typeInSearch('bar');
+
+    $items = $('.ember-power-select-options li');
 
     assert.equal($items.length, 2, 'finds searched model by label');
     assert.equal($items[0].innerText, 'bar', 'first item');
     assert.equal($items[1].innerText, 'bar second', 'second item');
-
-    done();
   });
-});
-
-test('Pre-loads content on preload=true', function(assert) {
-  this.render(hbs`{{model-picker updated=(action 'update')
-                                 model="foo"
-                                 preload=true
-                                 optionLabelPath="name"
-                                 searchProperty="name"}}`);
-
-  clickTrigger();
-
-  return wait()
-    .then(() => {
-      let $items = $('.ember-power-select-dropdown li');
 
-      assert.equal($items.length, 4, 'gets all objects of type on preload');
-      assert.equal($items[0].innerText, 'chad');
-      assert.equal($items[1].innerText, 'bar');
-      assert.equal($items[2].innerText, 'dave');
-      assert.equal($items[3].innerText, 'bar second');
+  test('Pre-loads content on preload=true', async function(assert) {
+    await render(hbs`{{model-picker updated=(action 'update')
+                                   model="foo"
+                                   preload=true
+                                   optionLabelPath="name"
+                                   searchProperty="name"}}`);
+
+    await clickTrigger();
+
+    return settled()
```

Comment:
I don't know what this is doing, but I think you can replace all the `then`-block with simple `await`, one of the benefits of this new testing method.